### PR TITLE
Remove `.html` suffix everywhere and make 404 page work on non-root directories

### PR DESCRIPTION
**Pull request by altrisi**
This PR removes the `.html` suffix from every page by moving them into their own folder, so they get disguised.

It also updates all the references in those pages to be absolute, that was painful.

Tested and it works (preview at https://altrisi.github.io/irisshaders.github.io, some links don't work properly there because it's not a root page).

Should I add a fallback in the old urls to redirect to the new ones, or is the 404 page enough?

I also fixed the 404 page not working at all when the not found page was not in the root directory, since it was useless in those situations before (it used relative references, therefore couldn't find anything in the same directory).

~(please squash don't leave this mess of commits live in history)~

### CODE REVIEW

**Comment by maxigator** [comment]
could you set up GitHub pages?

**Comment by altrisi** [comment]
Actually I could rename my repo, true, will try in a sec.

**Comment by altrisi** [comment]
Tested and it works (test url attached).

**Comment by maxigator** [comment]
there's an error but it's because the branch isn't synced with main

File: about/index.html [hunk 1]
```diff
@@ -37,29 +37,29 @@
...
-    link.href = "css/darkmode_supplements.css";
+    link.href = "/css/darkmode_supplements.css";
     link.type = "text/css";
     link.rel = "stylesheet";
     link.id = "darkThemeLink"
     document.getElementsByTagName( "head" )[0].appendChild( link );
   }
   document.addEventListener("DOMContentLoaded", function(event) {
-  	reloadTheme("about.html");
+  	reloadTheme("/about/");
```
Comment (maxigator) [inline about/index.html:47]:
```suggestion
  	reloadTheme("about.html");
```
don't change that

File: download/index.html [hunk 2]
```diff
@@ -37,29 +37,29 @@
...
-    link.href = "css/darkmode_supplements.css";
+    link.href = "/css/darkmode_supplements.css";
     link.type = "text/css";
     link.rel = "stylesheet";
     link.id = "darkThemeLink"
     document.getElementsByTagName( "head" )[0].appendChild( link );
   }
   document.addEventListener("DOMContentLoaded", function(event) {
-  	reloadTheme("download.html");
+  	reloadTheme("/download/");
```
Comment (maxigator) [inline download/index.html:47]:
```suggestion
  	reloadTheme("download.html");
```

**Review by maxigator — CHANGES_REQUESTED** [review]
you still have to keep thoses.
If not, you can edit the "fileindex" var in lang.js to be the same as you changed to
`var fileindex = ["index.html", "about.html", "download.html"];`

**Comment by thedocruby** [comment]
Why is this needed?

**Comment by altrisi** [comment]
It's just that IMO it looks better, and it's both easier to remember for end users and they can fully understand what's in the URL. Feel free to close it if you don't like it.

(this is https://irisshaders.net/about.html -> https://irisshaders.net/about/, https://irisshaders.net/download.html -> https://irisshaders.net/download/

**Comment by altrisi** [comment]
Wait actually don't merge this rn, just found that github provides a better way to do this.

**Comment by thedocruby** [comment]
Okay lmk if you need me to do anything

**Comment by altrisi** [comment]
Should be ready in a much less invasive way.

File: about.html [hunk 3]
```diff
@@ -43,22 +43,22 @@
     document.getElementsByTagName( "head" )[0].appendChild( link );
   }
   document.addEventListener("DOMContentLoaded", function(event) {
-  	reloadTheme("about.html");
+  	reloadTheme("about");
```
Comment (thedocruby) [inline about.html:46]: If you are going to change these then please also change to match in darkmode js file

**Review by thedocruby — COMMENTED** [review]
some extra work needs to be done

File: about.html (hunk 3 above)
Comment (altrisi) [reply about.html:46]: Assuming you mean lang.js file, since darkmode.js doesn't refer to those literals. (changed)

**Comment by maxigator** [comment]
going back to home returns a 404

**Pull request closed without merging**